implementation/common/src/main/java/io/smallrye/jwt/util/KeyUtils.java

```diff
@@ -246,6 +249,42 @@ public static SecretKey createSecretKeyFromSecret(String secret) {
         return new SecretKeySpec(secretBytes, "AES");
     }
 
+    /**
+     * Generates a SecretKey.
+     *
+     * @param algo key encryption algorithm.
+     * @return SecretKey.
+     * @throws NoSuchAlgorithmException algorithm not found.
+     */
+    public static SecretKey generateSecretKey(KeyEncryptionAlgorithm algo) throws InvalidAlgorithmParameterException {
+        final String algoName = getSymmetricAlgoName(algo);
```

Comment:
Here we only need to check if it is one of 3 `KW` algorithms and get the bits from the name

Reply:
You can safely drop `getSymmetricAlgoName` for now - even if it works within the `encryption` variation